src/components/List/src/ListItem.tsx

```diff
@@ -19,7 +19,8 @@ export interface ListItemProps extends BaseProps {
    * If `true`, the list item will be a button (using `ButtonBase`). Props intended
    * for `ButtonBase` can then be applied to `ListItem`.
    */
-  button?: false;
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  button?: any;
```

Comment:
This sucks, but I assume their is no other type that fits?

Reply:
It is a temporary fix, since when list-design is merged this property won't  be exposed anymore.